### PR TITLE
Add duration and progression_name to activity section

### PR DESCRIPTION
**Pull request by dmcavoy**
### Duration

In looking through a couple CSP lessons we noticed that times were being added in the title of activity sections. To better support this going forward we are adding a duration field to the activity section. (See image below)

### Progression Name

The name of the progression had been determined by the title of the activity section in order to keep fields down but this was not working for the way curriculum writers wanted to enter the information for activity sections so this adds a separate progression name field to activity section. The field only appears when you add a level to the activity section.

Screen Shot 2021-01-14 at 4 07 04 PM

### Level Hover Name

I also fixed a bug where the level name was showing when you hovered over the bubble on the lesson plan instead of the progression name. 

## Links

- [PLAT-664](https://codedotorg.atlassian.net/browse/PLAT-664)
- [PLAT-665](https://codedotorg.atlassian.net/browse/PLAT-665)
- [PLAT-669](https://codedotorg.atlassian.net/browse/PLAT-669)

## Testing story

Updated React Component Tests
- ActivitySectionCardTest.jsx 
- LessonEditorTest.js 
- ProgressionDetailsTest.jsx 

Updated Test Data
- activitiesTestData.js 

Updated the saving test
- lessons_controller_test.rb 

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/lessonOverview/activities/Activity.jsx [hunk 1]
```diff
@@ -21,9 +21,7 @@ export default class Activity extends Component {
     return (
       <div>
         <h2 style={styles.activityHeader} id={`activity-${activity.key}`}>
-          {i18n.activityHeaderName({
-            activityName: activity.displayName
-          })}
+          {activity.displayName}
```
Comment (dmcavoy) [inline apps/src/templates/lessonOverview/activities/Activity.jsx:24]: The i18n string was not doing anything so removed it.

File: apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js [hunk 2]
```diff
@@ -543,6 +543,10 @@ export const mapActivityDataForEditor = rawActivities => {
       activitySection.text = activitySection.description || '';
       delete activitySection.description;
 
+      activitySection.duration = activitySection.duration || '';
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js:546]: It looks like the default here is the empty string and below is 0. Is there a reason they're different?

**Review by bethanyaconnor — COMMENTED** [review]
One inline comment but overall the code change looks good.  However, I'm a bit concerned about the fact that just saving a lesson wipes out the progression names.

For examples, coursea-2021 lesson 5 goes from 
Screen Shot 2021-01-19 at 9 03 52 AM
to
Screen Shot 2021-01-19 at 9 04 20 AM
by just pressing save on the lesson edit page.

Right now, we might be able to include this in something the contractor looks at. We could also run a backfill on the existing data if that doesn't work.

File: apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js (hunk 2 above)
Comment (dmcavoy) [reply apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js:546]: Oh good call. I will update

**Comment by dmcavoy** [comment]
> One inline comment but overall the code change looks good. However, I'm a bit concerned about the fact that just saving a lesson wipes out the progression names.
> 
> For examples, coursea-2021 lesson 5 goes from
> Screen Shot 2021-01-19 at 9 03 52 AM
> to
> Screen Shot 2021-01-19 at 9 04 20 AM
> by just pressing save on the lesson edit page.
> 
> Right now, we might be able to include this in something the contractor looks at. We could also run a backfill on the existing data if that doesn't work.

@bethanyaconnor I was thinking about creating a migration to maybe set progression_name equal to title for now. That way we don't at least cause that regression and someone could go in and delete title or edit how they want. Thoughts?

**Comment by bethanyaconnor** [comment]
That sounds great!

**Comment by davidsbailey** [comment]
> @bethanyaconnor I was thinking about creating a migration to maybe set progression_name equal to title for now. That way we don't at least cause that regression and someone could go in and delete title or edit how they want. Thoughts?

@dmcavoy a migration could work, but it could be a good idea to write down a brief plan first, since this will be hard to undo.

a related question is, are we sure the curriculum writers want the progression name to be there? since part of the goal of this PR is to not show it twice, I guess i am wondering if we really want to set progression name equal to activity section title in so many places.

**Comment by dmcavoy** [comment]
> > @bethanyaconnor I was thinking about creating a migration to maybe set progression_name equal to title for now. That way we don't at least cause that regression and someone could go in and delete title or edit how they want. Thoughts?
> 
> @dmcavoy a migration could work, but it could be a good idea to write down a brief plan first, since this will be hard to undo.
> 
> a related question is, are we sure the curriculum writers want the progression name to be there? since part of the goal of this PR is to not show it twice, I guess i am wondering if we really want to set progression name equal to activity section title in so many places.

In general the current titles on activity sections that contain levels is the old progression name migrated over into this new system. I think what you are saying is valid tho and if we are going to copy the value it would be good to delete the activity section title so its not there twice. I think the challenge is the places where the levels have been moved and are no longer just in their own activity in the lesson. No matter what this is going to take a human eye on it

**Comment by davidsbailey** [comment]
makes sense. another piece of the puzzle is that i think we also store the progression name directly on the script level, for example this function gets called on lesson save and probably also got called on import: https://github.com/code-dot-org/code-dot-org/blob/9449933140cd7019517a5ec794460511eeef4267/dashboard/app/models/activity_section.rb#L79

**Comment by dmcavoy** [comment]
@davidsbailey @bethanyaconnor here is the pseudo code for what I'm thinking. I think the best plan is to just copy title to progression_name and not get rid of title. The contractors can delete the title if thats the correct thing to do.

```
For each migrated script
	For each lesson in the migrated script
		For each activity in the lesson
			For each activity section in activity
				if the activity section has script level
					set the progression_name to title
					save activity section
					for each script_level in activity_section
						set progression to progression_name
						save script level

```

**Comment by davidsbailey** [comment]
that sounds good to me, Dani. I see you've already handled the "update" codepath I linked to.

**Review by bethanyaconnor — APPROVED** [review]

**Review by davidsbailey — APPROVED** [review]
Looks good!

**Pull request merged**